src/Infrastructure/Dependencies.cs

```diff
@@ -30,11 +39,24 @@ public static void ConfigureServices(IConfiguration configuration, IServiceColle
             // Requires LocalDB which can be installed with SQL Server Express 2016
             // https://www.microsoft.com/en-us/download/details.aspx?id=54284
             services.AddDbContext<CatalogContext>(c =>
-                c.UseSqlServer(configuration["CATALOG_CONNECTION_STRING_VALUE"]));
+                c.UseSqlServer(catalogConnectionStringValue));
 
             // Add Identity DbContext
             services.AddDbContext<AppIdentityDbContext>(options =>
-                options.UseSqlServer(configuration["IDENTITY_CONNECTION_STRING_VALUE"]));
+                options.UseSqlServer(identityConnectionStringValue));
         }
     }
+
+    public static string GetSqlConnectString(string keyVaultUri, string connectionStringKey)
+    {
+        if (connectionStringKey == null)
+        {
+            return "";
+        }
+
+        var secretClient = new SecretClient(new Uri(keyVaultUri), new ClientSecretCredential("<tenant_id>","<client_id>","<client_secret>"));
+        KeyVaultSecret secret = secretClient.GetSecret(connectionStringKey);
```

Comment:
This should use DAC and also add the new AzureDeveloperCLICredential that can be found in azure-dev templates.  Look for the C# templates code on that.